core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -55,35 +60,44 @@
    private static final Log log = LogFactory.getLog(ClusterExpirationManager.class);
    private static final boolean trace = log.isTraceEnabled();
 
-   private static final int MAX_ASYNC_EXPIRATIONS = 5;
+   private static final int MAX_ASYNC_EXPIRATIONS = 100;
 
    @Inject protected ComponentRef<AdvancedCache<K, V>> cacheRef;
    @Inject protected ComponentRef<CommandsFactory> cf;
    @Inject protected RpcManager rpcManager;
    @Inject protected DistributionManager distributionManager;
 
    private AdvancedCache<K, V> cache;
+   private Address localAddress;
 
    @Override
    public void start() {
       super.start();
       // Data container entries are retrieved directly, so we don't need to worry about an encodings
       this.cache = AbstractDelegatingCache.unwrapCache(cacheRef.wired()).getAdvancedCache();
+      this.localAddress = cache.getCacheManager().getAddress();
    }
 
    @Override
    public void processExpiration() {
       long start = 0;
+      int removedEntries = 0;
```

Comment:
I'd like to extract the if body to a separate method to reduce indentation (and separate the data container purge from the persistence purge)

Reply:
Okay.